iceoryx_examples/request_response/README.md

```diff
@@ -0,0 +1,214 @@
+# request_response
+
+## Introduction
+
+This example demonstrates how to use iceoryx in a client-server architecture
+using the request-response communication pattern.
+
+## Expected output basic server-client example
+
+[![asciicast](https://asciinema.org/a/469913.svg)](https://asciinema.org/a/469913)
+
+## Code walkthrough
+
+In the following scenario the client (client_cxx_waitset.cpp) is using the `Waitset` to wait for a response from the server
+(server_cxx_listener.cpp) that uses the `Listener` API for taking and processing the requests.
+
+The client is inspired by the `iox-cpp-waitset-basic` example from the `waitset` folder and the server from the
+`iox-cpp-callbacks-subscriber` example in the `callbacks` folder.
+
+This is the most recommended way to create an efficient server-client combination with iceoryx.
+
+### Client using Waitset
```

Comment:
What is with the description for the client and server basic. Wouldn't it be better to present them first and then explain in the next example how one can attach them to a listener/waitset?

Reply:
The combination Server with Listener and Client using Waitset is the most natural setup, therefore the intention is to explain that more in detail. If we now explain additionally the basic setup then we also need to explain the untyped API in detail which would may make the Readme quite huge.
What do you think?

Reply:
I would at least note the other two examples in the introduction part, e. g. something like:
```We provide 3 examples, the very basic typed and untyped examples and the most natural setup combining a Server with a Listener and a Client using a Waitset. Since you can find the general setup and functionality of the client and the server also in the Listener/Waitset example, we will focus now on this.```